src/main/java/com/minecolonies/structures/helpers/Structure.java

```diff
@@ -82,63 +92,361 @@
                 {
-                    inputstream = new FileInputStream(decorationFolder.getPath() + "/" + structureName + ".nbt");
+                    return new FileInputStream(folder.getPath() + "/" + structureName + ".nbt");
                 }
                 else
                 {
-                    throw new FileNotFoundException("Unable to find structure: " + structureName);
+                    Log.getLogger().error("Structure: Illegal structure name \"" + structureName + "\"");
                 }
             }
-            catch (final FileNotFoundException e)
+            else
             {
-                Log.getLogger().warn("Couldn't find any structure with this name anywhere", e);
+                throw new FileNotFoundException("Unable to find structure: " + structureName);
             }
         }
+        catch (final FileNotFoundException e)
+        {
+            //Ignore the error
+        }
+        return null;
+    }
 
-        if (inputstream == null)
+    /**
+     * get a input stream for a schematic from jar.
+     *
+     * @param structureName name of the structure to load from the jar.
+     * @return the input stream or null
+     */
+    private static InputStream getStreamFromJar(final String structureName)
+    {
+        return MinecraftServer.class.getResourceAsStream("/assets/" + Constants.MOD_ID + '/' + structureName + ".nbt");
+    }
+
+    /**
+     * get the Template from the structure.
+     *
+     * @return The templae for the structure
+     */
+    public Template getTemplate()
+    {
+        return this.template;
+    }
+
+    /**
+     * Convert an InputStream into and array of bytes.
+     *
+     * @param stream to be converted to bytes array
+     * @return the array of bytes, array is size 0 when the stream is null
+     */
+    public static byte[] getStreamAsByteArray(final InputStream stream)
+    {
+        if (stream == null)
         {
-            return;
+            Log.getLogger().info("Structure.getStreamAsByteArray: stream is null this should not happen");
+            return new byte[0];
+        }
+        try
+        {
+            ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+
+            int nRead;
+            byte[] data = new byte[16384];
+
+            while ((nRead = stream.read(data, 0, data.length)) != -1)
+            {
+                buffer.write(data, 0, nRead);
+            }
+            return buffer.toByteArray();
+        }
+        catch (@NotNull IOException e)
+        {
+            Log.getLogger().trace(e);
         }
+        return new byte[0];
+    }
 
+    /**
+     * Calculate the MD5 hash for a template from an inputstream.
+     *
+     * @param stream to which we want the MD5 hash
+     * @return the MD5 hash string or null
+     */
+    public static String calculateMD5(final InputStream stream) //throws IOException
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ATrailingCommentCheck)